### DISCUSSION TITLE
CP on Single Node and Haproxy

### DISCUSSION THREAD

**Discussion by michallawson** [HCL-TECH-SOFTWARE/connections-automation#98] [Q&A]
Two questions:

1. In connections-automation/playbooks/setup-component-pack-complete-single-node.yml the Haproxy task is run twice. Is this correct?

2. Is Haproxy required where the customizer is installed on the master K8s; WAS and Connections are installed on another sever and NGINX proxy installed on a third server (ie small non-HA deployment)?

**Comment by dejo1307** [comment]
Hi Michael,

1. Yup, you are right, but that's not intentional. Will clean it up.
2. Haproxy is actually, in theory, not required on those small environments at all. If you check this playbook out - https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/main/playbooks/setup-component-pack-complete-development.yml - you will see it is exactly the same as the one you mentioned - just without Haproxy. The minimal environment can be simply two nodes with Customizer also enabled - one node with Nginx and Kubernetes, where Nginx is frontend door and there just to know what to send to IHS and what to send directly to Kubernetes, and the second one with DB, WAS, Connections etc.

Btw. that's a good point. Will add tomorrow a small how to with example how to add that minimal environment with just two nodes.

**Comment by michallawson** [comment]
Dejan,

Thank you.

Will try using that playbook, but will keep the WAS and DB2 on separate servers, so I just have three servers to deal with. 

Regards

Michael

**Reply by dejo1307** [reply]
So just a quick example: https://github.com/HCL-TECH-SOFTWARE/connections-automation/tree/main/environments/examples/cnx7/quick_start is using Haproxy. But if there is no Haproxy, then you just switch this variable here: https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/c9297a53dab06ea1d822952c4605419251f453f1/environments/examples/cnx7/quick_start/connections#L73 and instead of pointing it to where Haproxy lives, point it to the CP server.

You can also do it on already running system and circumvent Haproxy - just change the variable, and run playbooks/third_party/ibm-http-server-httpd-config.yml which would update httpd.conf with the new routes.

**Reply by michallawson** [reply]
Thank you.

Hope you do not mind one further query.

In the component pack environment file there are two variables as follows:

# If you are using different domains, this is needed to set up ingresses properly.
ce_on_prem=False
ingress_multi_domain_enabled=true

Am I correct in assuming that these should be changed to:

# If you are using different domains, this is needed to set up ingresses properly.
ce_on_prem=true
ingress_multi_domain_enabled=False

when all servers are in the same domain?

**Reply by dejo1307** [reply]
Actually, if it is all in the same domain, it doesn't matter, that domain will always be added to ingress, so even by default it can be set to true without anyone noticing. The only reason why it is not was that in the moment of opening this we were still testing older versions (and we still do) using this same automation, and this didn't exist until version 7 simply because Helm charts were different (the issue with OM giving 404 until you go and manually add also external domain there).
